app/src/main/java/com/thailam/piggywallet/ui/addtransaction/AddTransactionPresenter.java

```diff
@@ -0,0 +1,66 @@
+package com.thailam.piggywallet.ui.addtransaction;
+
+import android.support.annotation.NonNull;
+import android.widget.EditText;
+
+import com.thailam.piggywallet.data.model.Category;
+import com.thailam.piggywallet.data.model.Transaction;
+import com.thailam.piggywallet.data.source.TransactionDataSource;
+import com.thailam.piggywallet.util.Constants;
+
+public class AddTransactionPresenter implements AddTransactionContract.Presenter, TransactionDataSource.SaveTransactionCallback {
+    @NonNull
+    private AddTransactionContract.View mView;
+    @NonNull
+    private TransactionDataSource mTransactionRepository;
+
+    AddTransactionPresenter(@NonNull AddTransactionContract.View view,
+                            @NonNull TransactionDataSource transactionRepository) {
+        mView = view;
+        mTransactionRepository = transactionRepository;
+    }
+
+    @Override
+    public void start() {
+    }
+
+    @Override
+    public void saveTransaction(EditText editTextNote, EditText editTextAmount, Category category, long dateInput) {
```

Comment:
trong presenter k đc reference tới view
như vậy mới tăng khả năng testble
```suggestion
    public void saveTransaction(String node, String amount, Category category, long dateInput) {
```